src/equations/ideal_glm_mhd_multiion_2d.jl

```diff
@@ -0,0 +1,1113 @@
+with independent momentum and energy equations for each ion species. This implementation 
+assumes that the equations are non-dimensionalized such, that the vacuum permeability is ``\mu_0 = 1``.
+
+In case of more than one ion species, the specific heat capacity ratios `gammas` and the charge-to-mass 
+ratios `charge_to_mass` should be passed as tuples, e.g., `gammas=(1.4, 1.667)`.
+
+The argument `electron_pressure` can be used to pass a function that computes the electron
+pressure as a function of the state `u` with the signature `electron_pressure(u, equations::IdealGlmMhdMultiIonEquations2D)`.
+By default, the electron pressure is zero.
+
+The argument `initial_c_h` can be used to set the GLM divergence-cleaning speed. Note that 
+`initial_c_h = 0` deactivates the divergence cleaning. The callback [`GlmSpeedCallback`](@ref)
+can be used to adjust the GLM divergence-cleaning speed according to the time-step size.
+
+References:
+- G. Toth, A. Glocer, Y. Ma, D. Najib, Multi-Ion Magnetohydrodynamics 429 (2010). Numerical 
+  Modeling of Space Plasma Flows, 213–218.
+- A. Rueda-Ramírez, A. Sikstel, G. Gassner, An Entropy-Stable Discontinuous Galerkin Discretization
+  of the Ideal Multi-Ion Magnetohydrodynamics System (2024). Journal of Computational Physics.
+  [DOI: 10.1016/j.jcp.2024.113655](https://doi.org/10.1016/j.jcp.2024.113655).
+
+!!! ATTENTION: In case of more than one ion species, these equations should ALWAYS be used
+    with `source_terms_lorentz`.
+"""
+mutable struct IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT <: Real,
+                                              ElectronPressure} <:
+               AbstractIdealMhdMultiIonEquations{2, NVARS, NCOMP}
+    gammas::SVector{NCOMP, RealT} # Heat capacity ratios
+    charge_to_mass::SVector{NCOMP, RealT} # Charge to mass ratios
+    electron_pressure::ElectronPressure       # Function to compute the electron pressure
+    c_h::RealT                 # GLM cleaning speed
+    function IdealGlmMhdMultiIonEquations2D{NVARS, NCOMP, RealT,
+                                            ElectronPressure}(gammas
+                                                              ::SVector{NCOMP, RealT},
+                                                              charge_to_mass
+                                                              ::SVector{NCOMP, RealT},
+                                                              electron_pressure
+                                                              ::ElectronPressure,
+                                                              c_h::RealT) where
+             {NVARS, NCOMP, RealT <: Real, ElectronPressure}
+        NCOMP >= 1 ||
+            throw(DimensionMismatch("`gammas` and `charge_to_mass` have to be filled with at least one value"))
+
+        new(gammas, charge_to_mass, electron_pressure, c_h)
+    end
+end
+
+function IdealGlmMhdMultiIonEquations2D(; gammas, charge_to_mass,
```

Comment:
To reduce the complexity, can we implement this ctor in terms of the next one or vice versa?

Reply:
Right! Done in ffaf52f1533798e2cdf14a37497d5e9417c0507d.